src/adapter/etl-adapter-google-sheet/src/Flow/ETL/Adapter/GoogleSheet/GoogleSheetLoader.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Adapter\GoogleSheet;
+
+use Flow\ETL\{Adapter\GoogleSheet\RowsNormalizer\EntryNormalizer, FlowContext, Loader, Row\Entry, Rows};
+use Google\Service\Sheets;
+use Google\Service\Sheets\ValueRange;
+
+final class GoogleSheetLoader implements Loader, Loader\Closure
+{
+    private ?int $columnsCount = null;
+
+    private string $dateFormat = 'Y-m-d';
+
+    private string $dateTimeFormat = \DateTimeInterface::ATOM;
+
+    private int $startingRow = 0;
+
+    private string $timeFormat = 'H:i';
+
+    private bool $withHeader = true;
+
+    public function __construct(
+        private readonly Sheets $service,
+        private readonly string $spreadsheetId,
+        private readonly string $sheetName,
+    ) {
+    }
+
+    public function closure(FlowContext $context) : void
+    {
+        $this->startingRow = 0;
+        $this->columnsCount = null;
+    }
+
+    public function load(Rows $rows, FlowContext $context) : void
+    {
+        if (!$rows->count()) {
+            return;
+        }
+
+        if (null === $this->columnsCount) {
+            $this->columnsCount = $rows->first()->entries()->count();
+        }
+
+        $currentRow = $this->calculateCurrentRow($rows);
+        $sheetRange = $this->calculateSheetRange($currentRow);
+
+        $normalizer = new RowsNormalizer(
+            new EntryNormalizer($context->config->caster(), $this->dateTimeFormat, $this->dateFormat, $this->timeFormat)
+        );
+
+        $this->service->spreadsheets_values->update(
+            $this->spreadsheetId,
+            $sheetRange->toString(),
+            new ValueRange(
+                [
+                    'values' => $this->values($rows, $normalizer),
+                    'majorDimension' => 'ROWS',
+                ]
+            ),
+            ['valueInputOption' => 'RAW']
+        );
+
+        $this->startingRow = $currentRow;
+    }
+
+    public function withDateFormat(string $dateFormat) : self
+    {
+        $this->dateFormat = $dateFormat;
+
+        return $this;
+    }
+
+    public function withDateTimeFormat(string $dateTimeFormat) : self
+    {
+        $this->dateTimeFormat = $dateTimeFormat;
+
+        return $this;
+    }
+
+    public function withHeader(bool $withHeader) : self
+    {
+        $this->withHeader = $withHeader;
+
+        return $this;
+    }
+
+    public function withStartingRow(int $startingRow) : self
```

Comment:
so we can let user to decide to start writing from row 100, will it also put properly a header when custom starting row is set? 

Also if we want to allow users to set different starting row we should also let them set column different than A

Reply:
Actually, maybe we just let users to pass `SheetRange`?

Reply:
No, it was for internal testing purpose; it should not be exposed.

Reply:
at some point it might be a nice addition to let end users to change the starting address but it can come later once someone will get first use case for it